openlibrary/plugins/openlibrary/api.py

```diff
@@ -487,3 +488,35 @@ def response(msg, status="success"):
             }), content_type="application/json")
 
         return response('Observations removed')
+
+
+class works(delegate.page):
+    path = r"/works/OL(\d+)W"
+    encoding = "json"
+
+    def get_editions_of_work(self, work: Work) -> list[dict]:
+        keys: list = web.ctx.site.things({"type": "/type/edition", "works": work.key})
+        return web.ctx.site.get_many(keys, raw=True)
+
+    def DELETE(self, work_id: str):
+        user = accounts.get_current_user()
+        if not (user and (user.is_admin() or user.is_librarian())):
+            return web.HTTPError('403 Forbidden')
+
+        work_id = f"OL{work_id}W"
+        input = web.input(comment=None)
+
+        comment = input.get('comment')
+
+        work: Work = web.ctx.site.get(f'/works/{work_id}')
+        editions: list[dict] = self.get_editions_of_work(work)
+        keys_to_delete = [el.get('key') for el in [*editions, work.dict()]]
+        delete_payload = [
+            dict(key=key, type=dict(key='/type/delete'))
```

Comment:
This is a little easier for me to grok

```suggestion
            {'key': key, 'type': {'key': '/type/delete'}}
```